pinot-core/src/main/java/org/apache/pinot/core/data/manager/BaseTableDataManager.java

```diff
@@ -815,38 +814,49 @@ protected File downloadSegmentFromDeepStore(SegmentZKMetadata zkMetadata)
           segmentName, System.currentTimeMillis() - startTime, _segmentDownloadSemaphore.getQueueLength());
     }
     try {
-      File untarredSegmentDir;
-      if (_isStreamSegmentDownloadUntar && zkMetadata.getCrypterName() == null) {
-        _logger.info("Downloading segment: {} using streamed download-untar with maxStreamRateInByte: {}", segmentName,
-            _streamSegmentDownloadUntarRateLimitBytesPerSec);
-        AtomicInteger failedAttempts = new AtomicInteger(0);
-        try {
-          untarredSegmentDir = SegmentFetcherFactory.fetchAndStreamUntarToLocal(downloadUrl, tempRootDir,
-              _streamSegmentDownloadUntarRateLimitBytesPerSec, failedAttempts);
-          _logger.info("Downloaded and untarred segment: {} from: {}, failed attempts: {}", segmentName, downloadUrl,
-              failedAttempts.get());
-        } finally {
-          _serverMetrics.addMeteredTableValue(_tableNameWithType, ServerMeter.SEGMENT_STREAMED_DOWNLOAD_UNTAR_FAILURES,
-              failedAttempts.get());
+      if (_segmentOperationsThrottler != null) {
+        _segmentOperationsThrottler.getSegmentDownloadThrottler().acquire();
```

Comment:
You might want to add some log similar to the table level semaphore for debugging purpose

Reply:
done